### ISSUE TITLE
Some books fit inside plastic Bottles

### ISSUE THREAD

**Issue by AcidAntOnAMinefield** [CleverRaven/Cataclysm-DDA#55166]
### Describe the bug

Doing some reading, I noticed that my character would pick up and store the books inside empty bottles that I happened to have in my inventory. 
This apparently works with any container. Might it be related to the fact books don't seem to have length? 

### Steps to reproduce

1. Start a new world in a library/bookstore.
2. Debug yourself something hilarious like a 60L metal tank or a 200L Metal Drum.
3. Drag *all* the books into one tile.
4. [w]ield the container.
5. Go to your [i]nventory, choose the container and choose the [i]nsert option.
6. Grab all of the books.
7. Watch as your liquid tank becomes L-Space.

### Expected behavior

I wouldn't expect books to get into a liquids storage unit, or at least, not all books. 

### Screenshots

![image](https://user-images.githubusercontent.com/80117149/152701140-8e42d3ca-e05a-4717-88a2-6c775d7e5021.png)

![image](https://user-images.githubusercontent.com/80117149/152701144-fdd6fccd-b1fd-4351-8b67-feeaf58f842d.png)

### Versions and configuration

- Windows 7
- Experimental 0.F 2022-02-06-0843 d08dddd
- Tiles
- English
- Mods loaded: Dark Days Ahead [dda]

**Comment by martosss** [comment]
Bottles **can** store liquids, that's an **extra** additional bonus, not a constraint. You can also put non-liquid stuff in them.
The problem is that bottles are **rigid** - they keep their volume the same even if they're empty.
So your character, given the smart survivor he is, is using every possible free volume to store stuff when free volume is low.
That means, when you have no free space left in your bag, because it's full of empty bottles, you start putting stuff inside the bottles, where possible.

Think about a smart way to pack a travelling bag - you put socks inside your shoes and cutlery inside cups just because shoes and cups are **rigid* and there's empty space in them usually.

**tl;dr;**
Inventory management is smart and tries to fill every possible free pocket of your character.

**Comment by AcidAntOnAMinefield** [comment]
> Bottles can store liquids.
That is their whole purpose. 

However, the point is: Books. A 0.50L bottle is small. As in, 8cm in length according to the game. That's the size of your average computer mouse, give or take. There is no feasible way you would stick a Manual that outlines construction guidelines or gun maintenance techniques (going by those that are in the screenshot I posted) inside one, as one would assume said manuals, may be a little bit "longer" than the bottle itself including the ceiling on the cap. 

A pamphlet? Sure. A small instructions leaflet? Why not?
A book with several recipes and detailed instructions on things that will most likely require several pages worth of text to properly outline? Nah, don't think so.

**Comment by chrispikula** [comment]
I could see a few of the smaller, non-recipe containing manuals fitting inside a water bottle, but, yeah, stuff with recipes?  Not really.

And the manuals that would, would be the ones that raise your skill to 1 or, _maybe_, 2 in a category.  'How to change a tire', or the tiny booklets inside medkits things.  Pamphlets, in other words, like you said.

**Comment by a-chancey** [comment]
Actually, it looks like it's ignoring the max item volume tag on the bottle. 

The bottle shouldn't hold any individual item over 17ml, based on the json. 

Meanwhile both books are 250ml

I'm wondering if it has something to do with the container pocket being open.

**Comment by a-chancey** [comment]
Looking further: it's because paper is soft as a material, "True for pliable materials, whose length doesn't prevent fitting into a container, or through the opening of a container. Default is false."

one, the book sizes are just too small. They should be at least 10" (25cm) in length. Welding and metallurgy is only 12cm (less than 5"). 

Two, hardcover books should be mixed material, likely cardboard and paper. I didn't test this yet, but that *should* fix the issue. 

This should be an easy first issue

**Comment by a-chancey** [comment]
See my second comment. Open pockets are a feature in current experimental, but the reason rags fit is the same reason books do - they're made of a material that's flagged as being soft, so their length isn't taken into account for storage, just overall volume. 

Since an X-acto is steel and plastic, neither of which are soft, the length is used in the "will it fit" calculation. 

Relevant flag details are in docs\JSON_INFO.md
